pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -188,7 +197,7 @@ public long getLatestIngestionTimestamp() {
     // Initialize for each column
     for (FieldSpec fieldSpec : _physicalFieldSpecs) {
       String column = fieldSpec.getName();
-      _maxNumValuesMap.put(column, 0);
+      _numValuesInfoMap.put(column, new NumValuesInfo());
```

Comment:
Can we re-use the NumValuesInfo object instance by adding a clear method? I know we have been really careful about minimizing gc impact in this path.

Reply:
These are per segment objects, so no need to clear and reuse